build.gradle

```diff
@@ -83,13 +86,17 @@ repositories {
         allowInsecureProtocol true
         url "http://artifactory.terasology.org/artifactory/virtual-repo-live"
     }
+    maven {
+        // for spf4j dependencies that haven't been merged upstream
+        // i.e. org.apache.avro:avro
+        url  "https://dl.bintray.com/zolyfarkas/core"
+    }
     jcenter()
     maven { url "https://jitpack.io" }
 }
 
 // Primary dependencies definition
 dependencies {
-    implementation group: 'org.slf4j', name: 'slf4j-api', version: '2.0.0-alpha1'
```

Comment:
If the log backend needs it, it'll pull it in as a transitive dependency.

And once something *does* get in the `implementation` classpath it's real hard to get it out of the `testImplementation` classpath.

Reply:
hm, I think transitive dependencies are the exact thing that Java9+ modules tackle 🤔 You _can_ provide things transitively, but that's a decision the library has to make....
I gess that will be a lot of fun to figure out 🙈

Reply:
gradle works out transitive dependencies for packages too, but yeah, I guess it's different than what modules do. It looks like that's a big reason why these things are changing for slf4j 2.0, because they'll take advantage of the new module stuff instead of this stuff that makes for conflicts in classpath.

So it's possible I made the java 14 life more difficult by using something that's still on the java 8 style of things. Still hoping that since it's test-only it doesn't wreck your day when it comes to runtime and distribution dependencies.

Reply:
okay, I've updated with a syntax that manages to keep slf4j-api in as an explicit dependency . . .